server/src/main/java/org/opensearch/action/ActionModule.java

```diff
@@ -1111,17 +1111,25 @@ public void registerDynamicRoute(RestHandler.Route route, RestSendToExtensionAct
             requireNonNull(route, "route is required");
             requireNonNull(action, "action is required");
             Optional<String> routeName = Optional.empty();
+            Optional<String> legacyActionName = Optional.empty();
             if (route instanceof NamedRoute) {
                 routeName = Optional.of(((NamedRoute) route).name());
```

Comment:
nit: since `route` is being cast to `NamedRoute` in more than 1 place this may be more readable to create a variable:

`NamedRoute nr = (NamedRoute) route;`